java/fory-core/src/main/java/org/apache/fory/type/Descriptor.java

```diff
@@ -89,6 +89,8 @@ public static void clearDescriptorCache() {
   private final Method writeMethod;
   private ForyField foryField;
   private boolean nullable;
+  // trackingRef should only be true if explicitly set to true via @ForyField(ref=true)
+  // If no annotation or ref not specified, trackingRef stays false and type-based tracking applies
   private boolean trackingRef;
```

Comment:
We need to support disable ref tracking for a field by `@ForyField(ref=False)`. This can improve performance and reduce space overhead

Reply:
For `ref`:
1. if ref tracking is diabled globally, then ref is always false for all fields values
2. if ref tracking is enabled, then @ForyField has higher priority
3. otherwise, fallback to default ref tracking logic, use `needToWriteRef` instead

For `nullable`:
1. if `@ForyField` is configured, it has higher priority
2. if we support `NotNull/Nullable` annotation, thsi also has higher priority
3. othewise, fallback to default logic(only primitive types is not null).

Reply:
hm, there is an issue with collections: 
* when `useRefTracking` is `false` for the field (via `@ForyField()`), it writes the collection WITHOUT ref tracking wrapper
* but when deserializing, the `ArrayListSerializer.newCollection()` ALWAYS calls `fory.getRefResolver().reference(arrayList)` which expects ref tracking to be enabled

The issue is that newCollection() is being called regardless of whether ref tracking is enabled for this field or not.

So, I think we should give the global config the highest priority, hence the check in https://github.com/apache/fory/pull/3021#discussion_r2616725645 (`    boolean useRefTracking = explicitRefTracking || needWriteRef(typeRef);
`)is fine.

Reply:
@mchernyakov You can update `MapRefResolver` like this:

```java
  public void reference(Object object) {
    int refId = readRefIds.pop();
    if (refId >= 0) {
      setReadObject(refId, object);
    }
  }
```

Then for fields which will track ref but skip ref now, you can reserve an id `-1`:
```java
preserveRefId(-1)
```

`@ForyField` needs higher priority, it can improve performance signicantly. Most of times in an object graph, there is no shared/circular reference, but due to we can't differeiate those objects, we introduced lots of unnecesary ref tracking

Reply:
@chaokunyang addressed in https://github.com/apache/fory/pull/3021/commits/27e3280f715955f0ce60137affe7cab1ac80dad7 

and there is still question in https://github.com/apache/fory/pull/3021#discussion_r2617530972 

pls check